### PR TITLE
Terraform 0.14 upgrade

### PR DESCRIPTION
**Pull request by maximmi**
## what
- Upgrade to support Terraform 0.14 and bring up to current Cloud Posse standard

## why
- Support Terraform 0.14

### CODE REVIEW

**Comment by evilr00t** [comment]
Currently KMS version pinned is not supporting v0.14.15

```versions.tf line 2, in terraform:
   2:   required_version = ">= 0.12.0, < 0.14.0"

```

**Review by htplbc — APPROVED** [review]

**Review by osterman — APPROVED** [review]

File: .github/workflows/validate-codeowners.yml [hunk 1]
```diff
@@ -9,17 +9,10 @@ jobs:
     - name: "Checkout source code at current commit"
       uses: actions/checkout@v2
     - uses: mszostok/codeowners-validator@v0.5.0
-      if: github.event.pull_request.head.repo.full_name == github.repository
-      name: "Full check of CODEOWNERS"
```
Comment (osterman) [inline .github/workflows/validate-codeowners.yml:13]: This seems wrong. shouldn't be deleted

**Review by osterman — CHANGES_REQUESTED** [review]
This seems to be introducing a regression

**Comment by nitrocode** [comment]
This PR doesn't seem relevant anymore so Ill close this. Please reopen if I'm mistaken.

**Pull request closed without merging**